### ISSUE TITLE
Move affects wrong messages

### ISSUE THREAD

**Issue by Polfo** [Webklex/php-imap#64]
After using a query like ->whereSubject('process')
then iterating through the messages and moving them to a folder, 
different messages are moved.

```php
$folder = $this->client->getFolder('INBOX');
$messages = $folder->query()->setFetchOrder('desc')->whereSubject('ECoOB Process')->get();
foreach ($messages as $message) {
    $outs[] = '------------';
    $outs[] = $message->getDate() . ' ' . $message->getUid() .' '. $message->getSubject();
    if ($message->move('Destination', true) == true) {
        $outs[] = 'Message has been moved';
    } else { 
        $outs[] = 'Message could not be moved';
    }
}
return implode("<br>\n", $outs);
```

After executing code above, 
some messages with a different subject are moved, while other messages with the searched subject are still in the INBOX.
'Message could not be moved' appears for messages at the end of the list.

**Comment by Webklex** [comment]
Hi @Polfo ,
thanks for your report.
I assume `var_dump(ClientManager::get("options.sequence"));` returns 3? If this is the case, this means you are working with the message numbers which changes after each expunged action. You can either update `options.sequence` to `IMAP::FT_UID` or set expunge to false `Message::move("Destination", false)` and expunge at the end `Client::expunge()`.

Best regards,

**Comment by Polfo** [comment]
Yes `var_dump(ClientManager::get("options.sequence")); ` returns 3

I have set `options.sequence` to `IMAP::FT_UID`. That makes it worse: `move($destination, false)` and `move($destination, true)` always returns false and no message is moved.

I have also tested keeping the default `options.sequence` (3)
`$message->move("Destination", false)` moves some messages, but raise a `no headers found` exception
The messages moved do not match the ones on which `move()` is invoked.
For other message `move()` returns false. The number of not-moved matches the number of false returns.

`Message::move("Destination", true)` has the same behaviour.

I have tried simplifying the query to
`$messages = $folder->query()->all()->limit(10)->get();`
Same result.
